pkg/snapshot/layered_map.go

```diff
@@ -39,8 +46,18 @@ func NewLayeredMap(h func(string) (string, error)) *LayeredMap {
 func (l *LayeredMap) Snapshot() {
 	l.whiteouts = append(l.whiteouts, map[string]string{})
 	l.layers = append(l.layers, map[string]string{})
+	l.added = append(l.added, map[string]string{})
+}
+
+// Key returns a hash for added files
+func (l *LayeredMap) Key() (string, error) {
+	c := bytes.NewBuffer([]byte{})
+	enc := json.NewEncoder(c)
```

Comment:
It's not clear to me why we're json encoding anything. Is the idea to get a stable hash for a map[string]string?

I'm not sure if the sorting of a map when encoding it will be stable...